docs/pages/management/operations/self-hosted-automatic-agent-updates.mdx

```diff
@@ -0,0 +1,106 @@
+---
+
+<Details
+  title="Version warning"
+  opened={true}
+  scope={["enterprise"]}
+  scopeOnly={true}
+  min="13.0"
+>
+  Automatic agent update is available starting from Teleport `13.0`.
+</Details>
+
+Starting with Teleport 13, agents running in systemd or Kubernetes environments
+can be automatically updated. The [automatic updates architecture
+page](../../architecture/automatic-agent-update.mdx) describes how agent
+updating works.
+
+This guide covers how to set up the automatic update infrastructure. If this is
+already done, or you are a Teleport Cloud user, you can directly
+[enroll agents into automatic updates](./enroll-agent-into-automatic-updates.mdx).
+
+## Requirements
+
+- Self-hosted Teleport cluster running
+- `tctl` admin rights
+- either:
+  - a public S3 bucket, GCS bucket
+  - a webserver reachable from all agents with valid TLS certificates
+
+## Step 1/3 Create release channel files
+
+A release channel contains two pieces of information: the targeted version
+and if the update is critical. Updaters subscribe to a release channel and will
+update to the provided version if possible. If the update is critical, updaters
+will ignore the maintenance schedule and update as soon as possible.
+
+Create a directory for the new release channel `current`.
+
+```code
+$mkdir current/
+```
+
+Make the `current` release channel target the version (=teleport.version=):
+
+```code
+$ echo -n "(=teleport.version=)" > current/version
+```
+
+And mark the update as not critical:
+
+```code
+$ echo -n "no" > current/critical
+```
+
+## Step 2/3 Host the release channel files on the version server
+
+The release channel must be hosted on a webserver with trusted TLS certificates
+and reachable by all agents.
+
+Public cloud buckets like Amazon S3 or Google Cloud Storage are good
+candidates as they provide reliable low-maintenance file hosting.
+
+You can also serve the files with a regular webserver like `nginx`.
+The release channel needs to be served over HTTPS.
+
+Choose a way to serve the release channel and setup the hosting.
+
+The webserver must answer the following queries:
+```
+$ curl https://<hosting-domain-and-path>/current/version
+(=teleport.version=)
+
+$ curl https://<hosting-domain-and-path>/current/critical
+no
+```
+
+The web server serving the release channel is called the version server.
+Save the version server URL as it will be required later to configure the agent
+updaters.
+
+## Step 3/3 Configure the maintenance schedule
+
+At this point the updaters can be configured to pull the version from the
+release channel and update the agents. However, they still don't know when
+they should perform updates.
+
+Agents can retrieve the maintenance schedule from the Teleport cluster and
+pass it to the updater. In this step you'll configure the maintenance
+schedule for the whole cluster.
+
+Create the following `cmc.yaml` file to allow maintenances on Monday, Wednesday
+and Friday between 02:00 and 03:00 UTC.
```

Comment:
Please expand to explain where this file should reside, and/or how the Teleport cluster can/should incorporate it.

Reply:
done